compiler/test/dotty/tools/dotc/ParallelTesting.scala

```diff
@@ -0,0 +1,1076 @@
+package dotty
+package tools
+package dotc
+
+import java.io.{ File => JFile }
+import java.text.SimpleDateFormat
+import java.util.HashMap
+import java.lang.reflect.InvocationTargetException
+import java.nio.file.StandardCopyOption.REPLACE_EXISTING
+import java.nio.file.{ Files, Path, Paths, NoSuchFileException }
+import java.util.concurrent.{ Executors => JExecutors, TimeUnit, TimeoutException }
+
+import scala.io.Source
+import scala.util.control.NonFatal
+import scala.util.Try
+import scala.collection.mutable
+import scala.util.matching.Regex
+
+import core.Contexts._
+import reporting.{ Reporter, TestReporter }
+import reporting.diagnostic.MessageContainer
+import interfaces.Diagnostic.ERROR
+import dotc.util.DiffUtil
+
+/** A parallel testing suite whose goal is to integrate nicely with JUnit
+ *
+ *  This trait can be mixed in to offer parallel testing to compile runs. When
+ *  using this, you should be running your JUnit tests **sequentially**, as the
+ *  test suite itself runs with a high level of concurrency.
+ */
+trait ParallelTesting {
+
+  /** If the running environment supports an interactive terminal, each `Test`
+   *  will be run with a progress bar and real time feedback
+   */
+  def interactive: Boolean
+
+  /** A regex denoting which tests to run, please note that this gets turned
+   *  into an actual regular expression - and as such should be valid.
+   *
+   *  E.g: `*`is not a regular expression, but `.*` is.
+   */
+  def regex: Option[String]
+
+  /** The `regex` string gets turned into an actual regular expression used for
+   *  filtering tests
+   */
+  private lazy val filter: Option[Regex] = regex.map(str => new Regex(str))
+
+  /** A compilation target whose files or directory of files is to be compiled
```

Comment:
It's the target of the test - i.e. the files or directory currently being compiled. Do you have a constructive suggestion?

`TestTarget`?